frontend/ui/widget/button.lua

```diff
@@ -77,13 +79,51 @@ function Button:init()
     end
 
     if self.text then
+        local max_width = self.max_width and self.max_width - 2*self.padding_h - 2*self.margin - 2*self.bordersize or nil
         self.label_widget = TextWidget:new{
             text = self.text,
-            max_width = self.max_width and self.max_width - 2*self.padding_h - 2*self.margin - 2*self.bordersize or nil,
+            max_width = max_width,
             fgcolor = self.enabled and Blitbuffer.COLOR_BLACK or Blitbuffer.COLOR_DARK_GRAY,
             bold = self.text_font_bold,
             face = Font:getFace(self.text_font_face, self.text_font_size)
         }
+        self.did_truncation_tweaks = false
+        if self.avoid_text_truncation and self.label_widget:isTruncated() then
+            self.did_truncation_tweaks = true
+            local max_height = self.label_widget:getSize().h
+            local font_size_2_lines = TextBoxWidget:getFontSizeToFitHeight(max_height, 2, 0)
+            while self.label_widget:isTruncated() do
+                local new_size = self.label_widget.face.orig_size - 1
+                if new_size < font_size_2_lines then
+                    -- Switch to a 2-lines TextBoxWidget
+                    self.label_widget:free()
+                    self.label_widget = TextBoxWidget:new{
+                        text = self.text,
+                        line_height = 0,
+                        alignment = "center",
+                        width = max_width,
+                        height = max_height,
+                        height_adjust = true,
+                        height_overflow_show_ellipsis = true,
+                        fgcolor = self.enabled and Blitbuffer.COLOR_BLACK or Blitbuffer.COLOR_DARK_GRAY,
+                        bold = self.text_font_bold,
+                        face = Font:getFace(self.text_font_face, font_size_2_lines)
+                    }
+                    break
+                end
+                if new_size < 8 then -- don't go too small
```

Comment:
Should that be scaled?

Reply:
No, we're strill in unscaled territory there - it's then provided to Font:getFace(), and it is it that does the scaleBySize().